src/Sentry.Unity.Editor/Android/GradleDebugSymbolUpload.cs

```diff
@@ -0,0 +1,174 @@
+        {
+            foreach (string dirPath in Directory.GetDirectories(sourcePath, "*", SearchOption.AllDirectories))
+            {
+                Directory.CreateDirectory(dirPath.Replace(sourcePath, targetPath));
+            }
+
+            foreach (string newPath in Directory.GetFiles(sourcePath, "*.*", SearchOption.AllDirectories))
+            {
+                File.Copy(newPath, newPath.Replace(sourcePath, targetPath), true);
+            }
+        }
+
+        internal static string SetupSentryCli()
+        {
+            var sentryCliPath = Path.GetFullPath(Path.Combine("Packages", GetPackageName(), "Editor", "sentry-cli", GetSentryCli()));
+            if (!File.Exists(sentryCliPath))
+            {
+                throw new FileNotFoundException($"Could not find sentry-cli at path: {sentryCliPath}");
+            }
+
+            if (!SetPermissions(sentryCliPath))
+            {
+                // TODO: figure out what kind of exception is right here
+                throw new UnauthorizedAccessException($"Failed to set execute permissions on {sentryCliPath}");
+            }
+
+            return sentryCliPath;
+        }
+
+        internal static string GetPackageName()
+        {
+            var packagePath = Path.Combine("Packages", PackageName);
+            if (Directory.Exists(Path.Combine(packagePath)))
+            {
+                return PackageName;
+            }
+
+            packagePath = Path.Combine("Packages", PackageNameDev);
+            if (Directory.Exists(Path.Combine(packagePath)))
+            {
+                return PackageNameDev;
+            }
+
+            throw new FileNotFoundException("Failed to locate the Sentry package");
+        }
+
+        internal static string GetSentryCli()
+        {
+            return Application.platform switch
+            {
+                RuntimePlatform.WindowsEditor => "sentry-cli-Windows-x86_64.exe ",
+                RuntimePlatform.OSXEditor => "sentry-cli-Darwin-universal",
+                RuntimePlatform.LinuxEditor => "sentry-cli-Linux-x86_64 ",
+                _ => string.Empty
+            };
+        }
+
+        internal static bool SetPermissions(string sentryCliPath)
+        {
+            // TODO: do I need this? Can I make it smaller? _0755 = (int)493
+
+            // user permissions
+            const int S_IRUSR = 0x100;
+            const int S_IWUSR = 0x80;
+            const int S_IXUSR = 0x40;
+
+            // group permission
+            const int S_IRGRP = 0x20;
+            // const int S_IWGRP = 0x10;
+            const int S_IXGRP = 0x8;
+
+            // other permissions
+            const int S_IROTH = 0x4;
+            // const int S_IWOTH = 0x2;
+            const int S_IXOTH = 0x1;
+
+            const int _0755 =
+                S_IRUSR | S_IXUSR | S_IWUSR
+                | S_IRGRP | S_IXGRP
+                | S_IROTH | S_IXOTH;
+
+            if (access(Path.GetFullPath(sentryCliPath), (int)_0755) == 0)
```

Comment:
Not sure this is needed. Technically there's a race, since after checking this the access could have been changed anyway. And setting it to 755 when it's already 755 shouldn't be an issue and we avoid the double P/Invoke call